tiledb/sm/array/array.cc

```diff
@@ -64,10 +64,12 @@ Array::Array(const URI& array_uri, StorageManager* storage_manager)
     , storage_manager_(storage_manager) {
   is_open_ = false;
   array_schema_ = nullptr;
-  timestamp_ = 0;
+  timestamp_start_ = 0;
```

Comment:
Prefer using the member initialization list (e.g. L62-64) over assignment. This doesn't really matter for primitive types, but we should do this for all types as a best practice. The motivation is that using the member initialization list avoids unnecessarily invoking the default constructor for class types.

As you can here, there are places in the code base where we initialize member variables with assignment in the constructor's implementation. Typically when I add new member variables, I refactor the constructor so that all member variables are initialized in the member initialization list.

Please refactor this so that all members are initialized in the member initialization list.